cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/TransactionHttpHandler.java

```diff
@@ -175,4 +208,78 @@ public void resetTxManagerState(HttpRequest request, HttpResponder responder) {
     txClient.resetState();
     responder.sendStatus(HttpResponseStatus.OK);
   }
+
+  @Path("/transactions/pruneinfo")
+  @POST
+  public void getPruneInfo(HttpRequest request, HttpResponder responder) {
+    if (debugClazz == null || debugObject == null) {
+      responder.sendString(HttpResponseStatus.BAD_REQUEST, "Invalid List Pruning debug class not found.");
+      return;
+    }
+
+    try (Reader reader = new InputStreamReader(new ChannelBufferInputStream(request.getContent()))) {
+      String regionName = CharStreams.toString(reader);
+      Method method = debugClazz.getMethod("getRegionPruneInfo", String.class);
+      method.setAccessible(true);
+      Object response = method.invoke(debugObject, regionName);
+      if (response == null) {
+        responder.sendString(HttpResponseStatus.OK, "No prune upper bound has been registered for this region yet.");
```

Comment:
It would be good to keep the output as JSON when response code is 200. If we change the method to get, then we can return 404 when region is not found, right?